pkg/tfgen/installation_docs.go

```diff
@@ -124,3 +144,192 @@ func applyEditRules(contentBytes []byte, docFile *DocFile) ([]byte, error) {
+			}
+
+			// Prepare examples
+			// Make an example to record in the cliConverter.
+			fileName := fmt.Sprintf("configuration-installation-%d", i)
+			examples[fileName] = code
+
+			// Convert to PCL with convertViaPulumiCLI.
+			// This gives us a map of translatedExamples.
+			translatedExamples, err := g.cliConverter().convertViaPulumiCLI(examples, []tfbridge.ProviderInfo{
+				g.cliConverter().info,
+			})
+			if err != nil {
+				return "", err
+			}
+
+			// Write the result to the pcls map of our cli converter.
+			g.cliConverter().pcls[code] = translatedExample{
+				PCL: translatedExamples[fileName].PCL,
+			}
+
+			exPath := examplePath{fullPath: fileName}
+			conversionResult := g.coverageTracker.getOrCreateExample(
+				exPath.String(), code)
+
+			langs := genLanguageToSlice(g.language)
+			chooserStart := `{{< chooser language "typescript,python,go,csharp,java,yaml" >}}` + "\n"
+			returnContent = returnContent + chooserStart
+			chooserEnd := "{{< /chooser >}}\n"
+
+			// Generate each language in turn, so that we can mark up the output with the correct Hugo shortcodes.
+			// TODO: we want to use pulumi code choosers in the future, but resourcedocsgen does not support this yet.
+			for _, lang := range langs {
+				langSlice := []string{lang}
+				choosableStart := fmt.Sprintf("{{%% choosable language %s %%}}\n", lang)
+				choosableEnd := "\n{{% /choosable %}}\n"
+
+				// Generate the Pulumi.yaml config file for each language
+				configFile, err := writeConfigYamlViaCLI(outDir, lang, mappingsArgs)
+				if err != nil {
+					return "", err
+				}
+				// Generate example itself
+				convertedLang, err := g.convertHCL(conversionResult, code, exPath.String(), langSlice)
+				if err != nil {
+					return "", err
+				}
+				returnContent = returnContent + choosableStart + configFile + convertedLang + choosableEnd
+			}
+			returnContent += chooserEnd
+			startIndex = block.end + len(codeFence)
+		} else {
+			// Write any code block as-is.
+			returnContent = returnContent + contentStr[block.start:block.end+len(codeFence)]
+			startIndex = block.end + len(codeFence)
+		}
+	}
+	// Write any remainder.
+	returnContent = returnContent + contentStr[codeBlocks[len(codeBlocks)-1].end+len(codeFence):]
+	return returnContent, cleanupErr
+}
+
+// This function obtains the Pulumi.yaml config file for a given language.
+// It relies on the presence of a `main.tf` file for conversion.
+func writeConfigYamlViaCLI(outDir, lang string, mappingsArgs []string) (string, error) {
+	// Check if there's a main.tf file in the current working directory (outDir)
+	if _, err := os.Stat(filepath.Join(outDir, "main.tf")); err != nil {
+		return "", fmt.Errorf("expected main.tf file %w", err)
+	}
+	// Get the pulumi command
+	pulumiPath, err := exec.LookPath("pulumi")
+	if err != nil {
+		return "", fmt.Errorf("couldn't find pulumi path")
+	}
+
+	cmdArgs := []string{
+		"convert",
+		"--from",
+		"terraform",
+		"--language",
+		lang,
```

Comment:
Oof. You're converting from HCL to the target language directly via `pulumi convert`. We don't support this elsewhere in the bridge yet. Instead, we convert `TF->PCL` via `pulumi convert` and then we convert in-memory from PCL to JavaScript etc.